lib/rb_shift/project.rb

```diff
@@ -18,57 +18,86 @@ def self.open(name, client)
+      @_services
     end
 
-    def secrets
-      @_secrets ||= init_objects(Secret)
+    def secrets(update = false)
+      @_secrets = init_objects(Secret) if update || @_services.nil?
+      @_secrets
     end
 
-    def config_maps
-      @_config_maps ||= init_objects(ConfigMap)
+    def config_maps(update = false)
+      @_config_maps = init_objects(ConfigMap) if update || @_services.nil?
+      @_config_maps
     end
 
-    def templates
-      @_templates ||= init_objects(Template)
+    def templates(update = false)
+      @_templates = init_objects(Template) if update || @_templates.nil?
+      @_templates
     end
 
     def create_secret(name, kind, **opts)
-      `oc secrets #{kind} #{name} #{unfold_params(opts)}`
-      secrets << @client.get('secrets', :name => name, :namespace => @name) if @_secrets
+      execute "create secret #{kind} #{name}", **opts
+      secrets << @client.get('secrets', name: name, namespace: @name) if @_secrets
     end
 
     def create_template(file)
-      `oc create -f #{file}`
-      templates << @client.get('templates', :name => name, :namespace => @name) if @_templates
+      execute "create -f \"#{file}\""
+      templates << @client.get('templates', name: name, namespace: @name) if @_templates
     end
 
     def create_config_map(name, source, path, **opts)
-      `oc create configmap #{name} --#{source}=#{path} #{unfold_params(opts)}`
-      config_maps << @client.get('configmaps', :name => name, :namespace => @name) if @_config_maps
+      execute "create configmap #{name}", source.to_sym => path, **opts
+      config_maps << @client.get('configmaps', name: name, namespace: @name) if @_config_maps
     end
 
     def create_service(name, kind, **opts)
-      `oc create service #{kind} #{name} #{unfold_params(opts)}`
-      services << @client.get('services', :name => name, :namespace => @name) if @_services
+      execute "create service #{kind} #{name}", **opts
+      services << @client.get('services', name: name, namespace: @name) if @_services
+    end
+
+    def delete(block = false, timeout = 1)
+      execute "delete project #{@name}"
+      @client.wait_project_deletion(@name, timeout) if block
+    end
+
+    def wait_for_deployments(timeout = 30, update = false)
+      deployment_configs true
+      wait = true
+      while wait
+        sleep timeout
+        wait = !deployment_configs(update).select(&:running?).empty?
+      end
     end
 
     # Creates new Openshift application
     # Params:
     # +params+ - hash of key-value pairs to set/override a parameter value in the template
     # +args+ - any desired custom OC command options
-    def new_app(source, path, params = {}, **opts)
-      `oc new-app --#{source}=#{path} #{unfold_params(opts)} #{unfold_params(params, 'param')}`
-      invalidate
+    def new_app(source, path, block = false, timeout = 30, **opts)
+      execute 'new-app ', source.to_sym => path, **opts
```

Comment:
Well, you can't create symbol by colon from variable, for this you need to use the method
Source is variable which contains string name of source so there is to_sym method used properly.